core/trino-main/src/main/java/io/trino/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -297,6 +297,9 @@ else if (state == QueryState.FAILED) {
         if (queryStateMachine.isDone()) {
```

Comment:
@arhimondr I see we now have `DistributedStagesScheduler` and `PipelinedDistributedStagesScheduler`. Purpose of this classes and how they fit within scheduling is undocumented. Hence, it's hard for me to review `createDistributedStagesScheduler`. Could you maybe add some Javadocs for these classes.
It also seems that maybe `PipelinedDistributedStagesScheduler` (and the interface) should be a top level class

Reply:
> Could you maybe add some Javadocs for these classes.

Let me create a separate PR adding the docs.

> It also seems that maybe PipelinedDistributedStagesScheduler (and the interface) should be a top level class

This class is just an implementation detail of how `SqlQueryScheduler` operates, not sure if it makes sense to move it to the top level.

Reply:
https://github.com/trinodb/trino/pull/10289